portal/templates/initial_queries.html

```diff
@@ -844,7 +765,90 @@ <h4 class="profile-item-title"> {{ _("Thank you.") }}</h4>
     $(".button-container").each(function() {
       $(this).prepend('<div class="loading-message-indicator"><i class="fa fa-spinner fa-spin fa-2x"></i></div>');
     });
-    setTimeout('$("#iqFooterWrapper").show()', 2000);;
+
+    /*
+     * the flow here :
+     * get still needed core data
+     * populate all required fields
+     * get incomplete fields thereafter
+     * note: need to delay gathering incomplete fields to allow fields to be render
+     */
+    configObj.initConfig(function() {
+
```

Comment:
Would it make sense to move this section/functionality to a separate file, or does core data get acquired across multiple template/pages?

Reply:
Ivan, yes, it does, the code is growing.  I think I put the chunk there because of the dependency the code has on config variables, config.REQUIRED_CORE_DATA and config.CONSENT_WITH_TOP_LEVEL_ORG.  I know that breaking it out now into a separate file will break it.
I looked but don't think there is currently an API to get those values? If so, I can certainly break them out into a separate file.